daemon/daemon_unix.go

```diff
@@ -241,6 +271,16 @@ func verifyPlatformContainerSettings(daemon *Daemon, hostConfig *runconfig.HostC
 		logrus.Warnf("Your kernel does not support Block I/O weight_device. Weight-device discarded.")
 		hostConfig.BlkioWeightDevice = []*pblkiodev.WeightDevice{}
 	}
+	if len(hostConfig.BlkioReadBpsDevice) > 0 && !sysInfo.BlkioReadBpsDevice {
+		warnings = append(warnings, "Your kernel does not support Block read limit in bytes per second.")
+		logrus.Warnf("Your kernel does not support Block I/O read limit in bytes per second. Read-bps-device discarded.")
+		hostConfig.BlkioReadBpsDevice = []*pblkiodev.ThrottleDevice{}
+	}
+	if len(hostConfig.BlkioWriteBpsDevice) > 0 && !sysInfo.BlkioWriteBpsDevice {
+		warnings = append(warnings, "Your kernel does not support Block write limit in bytes per second.")
+		logrus.Warnf("Your kernel does not support Block I/O write limit in bytes per second. Write-bps-device discarded.")
```

Comment:
If we are discarding the device, shouldn't this be an error? Why is one message discarding and the other not from the same `if` statement ....sorry my Go knowledge is lame perhaps this is normal.